user/src/main/java/com/epam/esm/security/SecurityDecisionMaker.java

```diff
@@ -0,0 +1,44 @@
+package com.epam.esm.security;
+
+import com.epam.esm.dao.UserDao;
+import com.epam.esm.entity.User;
+import com.epam.esm.exception.ResourceNotFoundException;
+import org.keycloak.representations.AccessToken;
+import org.springframework.stereotype.Component;
+
+import java.util.Optional;
+
+/** The type Security decision maker. */
+@Component("authorizationDecisionMaker")
+public class SecurityDecisionMaker {
```

Comment:
I'd suggest to use more obvious naming for this bean, e.g. `CurrentAuthenticatedUserIdMatcher`/`CurrentUserIdMatcher`